src/vue/text-editor/text-editor-single-block.vue

```diff
@@ -83,17 +83,26 @@ export default {
     }
   },
   methods: {
-    textDirectionRadio (dir) {
+    directionRadioId (dir) {
       return `alpheios-alignment-editor-text-block__${this.textId}__${dir}`
     },
     updateAvaLang () {
       this.selectedOtherLang = null
       this.updateText()
     },
+    updateLang (lang) {
+      const langFromList = this.langsList.find(langOb => langOb.value === lang)
+
+      if (langFromList) {
+        this.selectedAvaLang = langFromList.value
+      } else {
+        this.selectedOtherLang = lang
+      }
+    },
     updateText () {
       this.$emit('update-text', {

```

Comment:
Why emit an event? Can't `this.$textC.updateTargetDocSource(text)` be called directly from here? Or is that for a reason to notify a parent component about the change (even though it does not seem to use this knowledge now)?

Reply:
it is done because I would like to place all business logic inside only main components - app, text-editor and align-editor (for now).
I would like to use practice to limit placing business logic inside visual components.

Reply:
Makes sense, I agree with this approach. Maybe just to add a comment here explaining where will it be handled?

Reply:
I believe that it is really a starting point of development and it would be changed several times, so it is too early time for detailed explanations inside code, from my point of view of course :)